src/System.Diagnostics.Process/src/System/Diagnostics/Process.Windows.cs

```diff
@@ -636,43 +658,71 @@ private bool StartCore(ProcessStartInfo startInfo)
 
                     if (startInfo.UserName.Length != 0)
                     {
+                        if(startInfo.Password != null && startInfo.PasswordInClearText != null)
+                        {
+                            throw new ArgumentException(SR.CantSetDuplicatePassword);
+                        }
+
                         Interop.mincore.LogonFlags logonFlags = (Interop.mincore.LogonFlags)0;
                         if (startInfo.LoadUserProfile)
                         {
                             logonFlags = Interop.mincore.LogonFlags.LOGON_WITH_PROFILE;
                         }
 
-
-                        try { }
-                        finally
+                        string password = string.Empty;
+                        IntPtr secureStringPtr = IntPtr.Zero;
+                        try
                         {
-                            retVal = Interop.mincore.CreateProcessWithLogonW(
-                                    startInfo.UserName,
-                                    startInfo.Domain,
-                                    startInfo.PasswordInClearText,
-                                    logonFlags,
-                                    null,            // we don't need this since all the info is in commandLine
-                                    commandLine,
-                                    creationFlags,
-                                    environmentPtr,
-                                    workingDirectory,
-                                    startupInfo,        // pointer to STARTUPINFO
-                                    processInfo         // pointer to PROCESS_INFORMATION
-                                );
+                            if(startInfo.Password != null)
+                            {
+                                secureStringPtr = Marshal.SecureStringToGlobalAllocUnicode(startInfo.Password);
+                                password = Marshal.PtrToStringUni(secureStringPtr);
```

Comment:
The reason why I created a regular string from it is because `Interop.mincore.CreateProcessWithLogonW` receives the password as a string, 
[Source Code](https://source.dot.net/#System.Diagnostics.Process/Common/Interop/Windows/Interop.CreateProcessWithLogon.cs,17). 

Do you have any suggestion in mind so that we can keep that Desktop behavior?